### ISSUE TITLE
Drag point changes after initial drag

### ISSUE THREAD

**Issue by tomdav999** [dimsemenov/PhotoSwipe#1196]
Is there a reason to remove the initial drag in function _renderMovement?  I think this is a bug.  To illustrate, position the mouse over image, click and drag quickly.  During and after dragging the mouse pointer should still be the initial drag point.  But it will change by the amount subtracted below.   

```
			if(_isFirstMove) {
				_isFirstMove = false;

				// subtract drag distance that was used during the detection direction  

				if( Math.abs(delta.x) >= DIRECTION_CHECK_OFFSET) {
					delta.x -= _currentPoints[0].x - _startPoint.x;
				}
				
				if( Math.abs(delta.y) >= DIRECTION_CHECK_OFFSET) {
					delta.y -= _currentPoints[0].y - _startPoint.y;
				}
			}
```

**Comment by dimsemenov** [comment]
I'm not sure if I can replicate the issue that you're describing. Can you please record some screencast or provide more details on how exactly to reproduce on http://photoswipe.com?

For the first ~20px PhotoSwipe doesn't move the image, as it detects the direction of swipe. As the image was not moved, but mouse was moved - the amount is substracted.

Thanks!

**Comment by tomdav999** [comment]
https://youtu.be/lIKoZSh9WGU

**Comment by tomdav999** [comment]
Here you go.  Note the position of the mouse cursor before drag and while dragging.  It is offset by the amount of initial drag.

https://youtu.be/lIKoZSh9WGU

**Comment by dimsemenov** [comment]
Please make the video Public or Unlisted, it's Private now, only you can view it.

Thanks!

**Comment by tomdav999** [comment]
Oops, hopefully you can see it now.  As far as the 20 pixel threshold, this test has occurred earlier when _direction was set, so by the time the code snippet above is reached we should drag the image by the distance the mouse has moved.  The initial drag is part of the drag so should not be subtracted.

**Comment by tomdav999** [comment]
OK, after playing around with this on touch vs. non-touch devices I think I better understand the issue.

First of all, the concept of an initial drag threshold is only pertinent to touch devices.  We need to differentiate between a drag and a tap, because fingers are not precicse.  If the user is using a mouse or precision pointer we shouldn't apply an initial drag threshold (because mice are precise).

Also, to the extent we have a minimum initial drag threshold, we should subtract the initial drag for smoothness.  This is what the code is trying to do, but it actually isn't smooth because it only subtracts the initial drag in *ONE* direction.  For smoothness it should subtract the initial drag in *BOTH* directions.  Also, the initial drag threshold should be checked against the distance dragged not the direction dragged.  Remember, we are attempting to compensate for clumsy fingers, not force the user to drag in any one direction!   Note the current code makes it very difficult to drage at an angle (when zoomed in).

Here is the correct way (in my humble opinion) to handle this.

//	if(Math.abs(diff) >= DIRECTION_CHECK_OFFSET) {
//
// replace the line commented above with this code (so that the user can drag at any angle when zoomed in):
//
	DIRECTION_CHECK_OFFSET = Math.abs(DIRECTION_CHECK_OFFSET);
	if(e.type.indexOf('mouse') > -1 || e.pointerType && e.pointerType.indexOf('mouse') > -1) {
		DIRECTION_CHECK_OFFSET = -DIRECTION_CHECK_OFFSET; // do not appply threhold for mice
	}
	var dist = Math.sqrt(Math.pow(touchesList[0].x - _currPoint.x, 2) + Math.pow(touchesList[0].y - _currPoint.y, 2));
	if(dist >= DIRECTION_CHECK_OFFSET) {
.
.
.
if(_isFirstMove) {
	_isFirstMove = false;

	// subtract drag distance that was used during the detection direction
//	if( Math.abs(delta.x) >= DIRECTION_CHECK_OFFSET) {
//		delta.x -= _currentPoints[0].x - _startPoint.x;
//	}
//	if( Math.abs(delta.y) >= DIRECTION_CHECK_OFFSET) {
//		delta.y -= _currentPoints[0].y - _startPoint.y;
//	}
//
// replace the code commented above with this code (so that drag is smooth when dragging at an angle):
//
	if(DIRECTION_CHECK_OFFSET > 0) {
		delta.x -= _currentPoints[0].x - _startPoint.x;
		delta.y -= _currentPoints[0].y - _startPoint.y;
	}

**Comment by dimsemenov** [comment]
I agree with most of your statements. The fix will be included in the next major update. Thanks!
